lib/LaTeXML/Engine/LaTeX.pool.ltxml

```diff
@@ -1383,8 +1383,11 @@ sub applyAligningContext {
   if (my $container = LookupValue('ALIGNING_NODE')) {
     my ($node, $previous) = @$container;
     my @children = $node->childNodes;
-    while (my $skip = shift(@children)) {
-      last if !$previous || $previous->isSameNode($skip); }
+    while (@children) {
+      my $skip = $children[0];
+      last if !$previous;
```

Comment:
since `$previous` is not changed during this `while` loop, maybe the check that it's defined can be lifted up? As in:

```perl
if ($previous) {
  while (my $skip = shift(@children)) {
    last if $previous->isSameNode($skip);
  }
}
```